spec/controllers/users/phone_setup_controller_spec.rb

```diff
@@ -234,6 +234,41 @@
         expect(subject.user_session[:context]).to eq 'confirmation'
       end
     end
+
+    context 'with rate phone fingerprint rate limit' do
+      before do
+        @user = create(:user)
+        @user2 = create(:user)
+        @unconfirmed_phone = '+1 (202) 555-1213'
+        @unconfirmed_phone2 = '+1 (202) 555-1215'
+      end
+      it 'displays an error banner' do
+        sign_in_before_2fa(@user)
+        allow(IdentityConfig.store).to receive(:otp_delivery_blocklist_maxretry).and_return(999)
+        allow(IdentityConfig.store).to receive(
+          :phone_submissions_per_fingerprint_max_attempts_window_in_minutes,
+        ).and_return(10)
```

Comment:
We could make the test run a little faster if we reduced this a bit so it's not looping as many times in the loop that follows.

```suggestion
        ).and_return(2)
```